src/rng.rs

```diff
@@ -137,6 +137,24 @@ mod test {
         }
     }
 
+    impl rand_crypto::RngCore for TestRng {
+        fn next_u32(&mut self) -> u32 {
+            self.0.next_u32()
+        }
+
+        fn next_u64(&mut self) -> u64 {
+            self.0.next_u64()
+        }
+
+        fn fill_bytes(&mut self, dest: &mut [u8]) {
+            self.0.fill_bytes(dest)
+        }
+
+        fn try_fill_bytes(&mut self, _dest: &mut [u8]) -> Result<(), rand_crypto::Error> {
+            Ok(())
```

Comment:
Probably need to call `self.0.fill_bytes(dest)` before returning Ok, this should not have stub non functional implementation.

Reply:
Does `self.0.fill_bytes(dest)` not return the correct Result? I mean should that not be the last line with no `;` ? @maqi

Reply:
`fill_bytes` returns `()`, but we need to return `Result`.

Reply:
We can do something like `Ok(self.0.fill_bytes(dest))`
but I think it's bit mis-leading?

Reply:
Yes, it would be misleading.